tests/test_queries.py

```diff
@@ -0,0 +1,890 @@
+    """Base class for all test queries"""
+
+    stats = None
+
+    def __init__(self, registry_path, project, output_dir=Path("queries")):
+        self._registry_path = Path(registry_path)
+        self._project = project
+        self._output_dir = Path(output_dir)
+        self._model = None
+        self._spark = SparkSession.builder.appName("dgrid").getOrCreate()
+
+    @property
+    def name(self):
+        """Return the name of the query.
+
+        Returns
+        -------
+        str
+
+        """
+        return self.make_query().name
+
+    @property
+    def output_dir(self):
+        """Return the output directory for the query results.
+
+        Returns
+        -------
+        Path
+
+        """
+        return self._output_dir
+
+    def get_raw_stats(self):
+        """Return the raw stats for the data tables.
+
+        Returns
+        -------
+        dict
+        """
+        if QueryTestBase.stats is None:
+            logger.info("Generate raw stats")
+            QueryTestBase.stats = generate_raw_stats(self._registry_path)
+        return QueryTestBase.stats
+
+    @abc.abstractmethod
+    def make_query(self):
+        """Return the query model"""
+
+    @abc.abstractmethod
+    def validate(self):
+        """Validate the results
+
+        Returns
+        -------
+        bool
+            Return True when the validation is successful.
+
+        """
+
+    def get_filtered_county_id(self):
+        filters = self._model.project.dimension_filters
+        counties = [x.value for x in filters if x.query_name == "county"]
+        assert len(counties) == 1, f"Unexpected length of filtered counties: {len(counties)}"
+        return counties[0]
+
+    def get_expected_ca_max_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return max(
+            (
+                by_county["06037"]["comstock_resstock"]["max"]["electricity"],
+                by_county["06073"]["comstock_resstock"]["max"]["electricity"],
+            )
+        )
+
+    def get_expected_ny_max_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return max(
+            (
+                by_county["36047"]["comstock_resstock"]["max"]["electricity"],
+                by_county["36081"]["comstock_resstock"]["max"]["electricity"],
+            )
+        )
+
+    def get_expected_ca_sum_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return (
+            by_county["06037"]["comstock_resstock"]["sum"]["electricity"]
+            + by_county["06073"]["comstock_resstock"]["sum"]["electricity"]
+        )
+
+    def get_expected_ny_sum_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return (
+            by_county["36047"]["comstock_resstock"]["sum"]["electricity"]
+            + by_county["36081"]["comstock_resstock"]["sum"]["electricity"]
+        )
+
+
+class QueryTestElectricityAgg(QueryTestBase):
```

Comment:
I used this order because I want to see the tests at the top of the file. I run one test at a time while developing and this helps me find them quickly. Let me know if you still disagree.